src/modules/m_spanningtree/main.cpp

```diff
@@ -814,8 +813,15 @@ ModuleSpanningTree::~ModuleSpanningTree()
 {
 	ServerInstance->PI = &ServerInstance->DefaultProtocolInterface;
 
-	Server* newsrv = new Server(ServerInstance->Config->GetSID(), ServerInstance->Config->ServerName, ServerInstance->Config->ServerDesc);
-	SetLocalUsersServer(newsrv);
+	try
+	{
+		Server* newsrv = new Server(ServerInstance->Config->GetSID(), ServerInstance->Config->ServerName, ServerInstance->Config->ServerDesc);
+		SetLocalUsersServer(newsrv);
+	}
+	catch (const std::bad_alloc& e)
```

Comment:
Trying to log here would just rethrow bad_alloc as the logger allocates memory too. If there's no more memory then there's not really much we can do other than exit so handling `bad_alloc` is unnecessary.

Reply:
@SadieCat, fix it.
Commit info here:
https://github.com/inspircd/inspircd/pull/1879/commits/b4e0c5b6212def02386af0a555445412eed6f1f4